tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -226,16 +232,277 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
       emitNextLine(0)
   }
 
+  // Private API for Rhino
+
+  private[scalajs] object rhinoAPI { // scalastyle:ignore
+    private implicit val GlobalKnowledge: GlobalKnowledge = new Invalidatable {
+      override def invalidate(): Unit = ()
+    }
+
+    def initialize(linkingUnit: LinkingUnit): Unit =
+      startRun(linkingUnit)
+
+    def getHeaderFile(): org.scalajs.core.tools.io.VirtualJSFile =
+      CoreJSLibs.lib(semantics, outputMode)
+
+    def genClassDef(linkedClass: LinkedClass): js.Tree =
+      classEmitter.genClassDef(linkedClass)
+  }
+
+  // Guardian of all global knowledge
+
+  private abstract class InvalidatableBag {
```

Comment:
Could we use the same terminology as in the optimizer? IIUC this is called `Unregisterable` in the optimizer (which I find easier to understand).

Further, the meaning of "register" (in `registerBag`) seems to be inversed. In the optimizer, an `Invalidatable`, i.e. a `MethodImpl` can register to a dependency and is itself `registeredTo` and can be `unregistered`.

Here we use the terminology `register` as `registeredTo` and `forget` for unregister.

TBH, I'm wondering if it is worth making a common base type `Unregisterable[T]` for both the optimizer and here (maybe also for `Invalidatable`)? (This makes me think that `invalidate` is called `tag` in the optimizer. `invalidate` is definitely the better name, so we should keep it and maybe change it in the optimizer).